addons/namingng.py: Fix commandline use.

thingsconnected opened the pull request:
namingng.py was only usable in standalone mode, but now supports CLI mode, i.e. with cppcheck --addon=namingng. It uses the generic reporting provided by cppcheckdata.reportError(). All output other than reported errors is suppressed.

A local function reportNamingError() is implemented to call through to cppcheckdata.reportError(), filling in common defaults.

The collection of errors and the --verify feature are removed, including related workflow and a test file. These are replaced by a unit test.

danmar commented on addons/namingng.py:
```diff
@@ -28,78 +28,70 @@
 
 def process(dumpfiles, configfile, debugprint=False):
-
-    errors = []
```
I wonder is there a special reason to make errors global.

thingsconnected replied:
Not really, but passing around `errors` all the way down to the point where errors are finally logged, felt a bit pointless in a script like this. I actually think collecting errors might also be removed from the code; the verification can be done in a different way, comparing the output as it is produced by the script, and then the only reason left to collect errors is to print the line "Found errors: N" - is that line important to keep?

danmar replied:
> I actually think collecting errors might also be removed from the code

Yes it's really clumpsy/ugly to hard code the expected test output in this script. so we should do something else. We have some tests that utilize the inline suppressions instead. Okay.. if our plan is to get rid of errors anyway I don't have to worry about this detail..

> and then the only reason left to collect errors is to print the line "Found errors: N" - is that line important to keep?

I don't feel that is important.

danmar reviewed (commented):
thanks for giving this addon a bit of attention

thingsconnected commented:
I got feedback in a different PR that a unit test should be implemented, so I did. To test `namingng`, the RE_FILE feature had to be fixed as well so this PR now contains a very minor fix (https://github.com/danmar/cppcheck/pull/5802 is an improved version of it).

danmar reviewed (commented):
if you test the addon properly with test-other.py then I do not feel that the `--verify` option is needed and then we don't need to return errors from reportError..

thingsconnected commented:
> if you test the addon properly with test-other.py then I do not feel that the `--verify` option is needed and then we don't need to return errors from reportError..

I did a cleanup and put all of it in one commit, as all changes are intertwined. The test case is not as elaborate as the `--verify` variant on some details, but does cover (include) file naming tests. I don't think every single violation of a pattern needs testing, rather just the concept of testing naming of filenames, function names and variable names, plus the prefix test feature. Please let me know if more tests should be added.

Furthermore, I added a commit from a different pending PR, regarding the naming of the default namingng config file. I also added a `namingng.json` example to that commit, to save new users the time to figure that one out.

firewave commented:
Thanks for working on this. I also ran into some of the issues with the addon and also filed https://trac.cppcheck.net/ticket/12005 about it not being working. I guess that ticket is now fixed.

I was considering to deprecate this addon as it was never maintained or tested at all after it was upstreamed (this applies to several addons). That's the reason I didn't post several changes I made to get it working. I am looking at them right now if there is still something to salvage.

As this addon is now getting more attention I wonder if we should deprecate the `naming` addon or replace it with the other one. Having two ways of doing the same thing doesn't make much sense.

firewave commented on .github/workflows/CI-unixish.yml:
```diff
@@ -446,8 +446,6 @@ jobs:
           python3 ../naming.py --var='[a-z].*' --function='[a-z].*' naming_test.c.dump
           ../../cppcheck --dump naming_test.cpp 
           python3 ../naming.py --var='[a-z].*' --function='[a-z].*' naming_test.cpp.dump
-          ../../cppcheck --dump namingng_test.c 
```
We should at least keep a basic test to make sure it still works in standalone mode.

thingsconnected commented:
> As this addon is now getting more attention I wonder if we should deprecate the `naming` addon or replace it with the other one. Having two ways of doing the same thing doesn't make much sense.

I could simply implement `--var=`, `--const=`, `--private-member-variable=` and `--function=` in `namingng.py` (only for non-CLI mode) and it would be a drop-in replacement for `naming.py`, meaning we could then safely rename `namingng.py` to `naming.py` with nobody noticing.
